pkg/operator/csidriveroperator/driverstarter.go

```diff
@@ -159,16 +163,27 @@ func (c *CSIDriverStarterController) sync(ctx context.Context, syncCtx factory.S
 func (c *CSIDriverStarterController) createCSIControllerManager(
 	cfg csioperatorclient.CSIOperatorConfig,
 	clients *csoclients.Clients,
-	resyncInterval time.Duration) manager.ControllerManager {
+	resyncInterval time.Duration) (manager.ControllerManager, func() []configv1.ObjectReference) {
 
 	manager := manager.NewControllerManager()
-	manager = manager.WithController(staticresourcecontroller.NewStaticResourceController(
+	var ctrlRelatedObjects func() []configv1.ObjectReference
+
+	src := staticresourcecontroller.NewStaticResourceController(
 		cfg.ConditionPrefix+"CSIDriverOperatorStaticController",
-		generated.Asset,
-		cfg.StaticAssets,
-		resourceapply.NewKubeClientHolder(clients.KubeClient),
-		c.operatorClient,
-		c.eventRecorder).AddKubeInformers(clients.KubeInformers), 1)
+		generated.Asset, cfg.StaticAssets, resourceapply.NewKubeClientHolder(clients.KubeClient), c.operatorClient, c.eventRecorder).
+		AddKubeInformers(clients.KubeInformers).
+		AddRESTMapper(clients.RestMapper)
+
+	manager = manager.WithController(src, 1)
+	objs, err := src.RelatedObjects()
+	if err != nil {
+		// TODO : error msg or aggregate error
+		klog.Errorf("Can not find RelatedObjects, got %v", err)
+	} else {
+		ctrlRelatedObjects = func() []configv1.ObjectReference {
+			return objs
+		}
+	}
```

Comment:
Please move `src.RelatedObjects` call to returned `func() ([]configv1.ObjectReference, error)` and call it during sync() - it can have proper error handling there and automatic retry on random API server error.

Reply:
RelatedObject( ) can be only called from StaticResourceController .. Since we use type ControllerManager within csiDriverControllerManager , its not possible to call it within Sync()

Reply:
actually thats why we added the funktor, so it has the related objects within the scope of `Sync()` and then we get the data from it, afterwards the funktor it self is `garbage` collected

Reply:
```
ctrlRelatedObjects = func() ([]configv1.ObjectReference, error) {
			return src.RelatedObjects()
		}
```

Reply:
Ahh I c, I thought u want everything in Sync .. ok On it 👍🏽

Reply:
@jsafrane Done 👍🏽